doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,635 @@
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Overview](#overview)
+  - [VLAN Stacking Deployment Use Cases](#vlan-stacking-deployment-use-cases)
+    - [QinQ Use Case](#qinq-use-case)
+    - [VLAN Translation Use Case](#vlan-translation-use-case)
+- [Architecture Design](#architecture-design)
+- [Requirements](#requirements)
+  - [Configuration](#configuration)
+- [High-Level Design](#high-level-design)
+  - [VlanMgr](#vlanmgr)
+  - [DB](#db)
+    - [CONFIG_DB](#config_db)
+    - [APPL_DB](#appl_db)
+  - [SAI API](#sai-api)
+- [Configuration and management](#configuration-and-management)
+  - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+    - [CLI](#cli)
+    - [Yang model](#yang-model)
+  - [Examples](#examples)
+    - [Examples of QinQ Configuration](#examples-of-qinq-configuration)
+    - [Examples of VLAN Translation Configuration](#examples-of-vlan-translation-configuration)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictionslimitations)
+- [Testing Requirements/Design](#testing-requirementsdesign)
+  - [System Test Cases](#system-test-cases)
+
+# List of Tables
+
+* [Table 1: Revision](#revision)
+* [Table 2: Definitions/Abbreviations](#definitionsabbreviations)
+* [Table 3: SAI attributes related to VLAN stacking](#sai-api)
+
+# Revision
+
+###### Table 1: Revision
+
+| Rev   |   Date     | Author        | Change Description |
+|:-----:|:----------:|:-------------:|--------------------|
+| 0.1   | 8-Oct-2020 | Tommy Tseng   | Initial version    |
+
+# Scope
+
+This document describes the high level design of VLAN stacking feature.
+
+# Definitions/Abbreviations
+
+###### Table 2: Definitions/Abbreviations
+
+| Abbreviation | Full form              |
+|--------------|------------------------|
+| VID          | Virtual LAN Identifier |
+| S-VLAN       | Service Provider VLAN  |
+| C-VLAN       | Customer VLAN          |
+| S-TAG        | S-VLAN tag             |
+| C-TAG        | C-VLAN tag             |
+
+# Overview
+
+This document describes the design details of the VLAN stacking feature. VLAN stacking is a feature designed for service providers who carry traffic of multiple customers across  their networks and are required to maintain the VLAN and Layer 2 protocol configurations of each customer without impacting the traffic of other customers. VLAN stacking include QinQ feature and VLAN Translation feature.
+
+IEEE 802.1Q tunneling (QinQ tunneling) uses a single Service Provider VLAN (SPVLAN) for customers who have multiple VLANs. Customer VLAN IDs are preserved and traffic from different customers is segregated within the service provider’s network even when they use the same customer-specific VLAN IDs. QinQ tunneling expands VLAN space by using a VLAN-in-VLAN hierarchy, preserving the customer’s original tagged packets, and adding SPVLAN tags to each frame (also called double tagging).
+
```

Comment:
Is MAC-learning targeted in this feature ? If yes, are MACs learned in context of C-VLAN or S-VLAN. If we are learning in S-VLAN context then MACs have to unique across C-VLANs.

Reply:
Yes, MAC-learning works in this feature, when the feature is configured, the MAC will learn in S-VLAN. Will write new comments on this HLD.

Reply:
@sunesh Is there any other comment there, ifnot, please help to approve to resolve this.

Reply:
@pttch may be later in the doc u can clarify n:1 map for C-VLAN to S-VLAN requires macs to be unique across all C-VLANs since learning is happening only on S-VLAN.